### PR TITLE
integration test

### PR DESCRIPTION
**Pull request by MartinWeindel**
**How to categorize this PR?**

/area dev-productivity
/kind test
/priority normal
/platform vsphere

**What this PR does / why we need it**:
Added integration tests for the infrastructure controller.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement developer
A new integration test for infrastructure creation and deletion has been added.
```

### CODE REVIEW

**Review by dguendisch — APPROVED** [review]
After having this merged, we need to include/add it to the vsphere cron TestRun.

**Review by dkistner — COMMENTED** [review]
looks good from my perspective.

I can't judge if all necessary cases are covered as I'm not a vsphere expert.
Maybe @mandelsoft could have a look too?

File: pkg/vsphere/infrastructure/task/tasks.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func (t *tier1GatewayLocaleServiceTask) ListAll(ctx EnsurerContext, state *api.N
 
 func (t *tier1GatewayLocaleServiceTask) EnsureDeleted(ctx EnsurerContext, state *api.NSXTInfraState) (bool, error) {
 	if t.IsExternal(state) {
-		return true, nil
+		return false, nil
```
Comment (mandelsoft) [inline pkg/vsphere/infrastructure/task/tasks.go:366]: Why?

Comment (MartinWeindel) [reply pkg/vsphere/infrastructure/task/tasks.go:366]: This was a minor inconsistency. The returned boolean flags if the resource has been deleted.

**Review by mandelsoft — APPROVED** [review]

**Pull request merged**